server/graphql/v1/mutations/orders.js

```diff
@@ -641,65 +642,61 @@ export async function completePledge(remoteUser, order) {
       message: 'You need to be logged in to cancel a subscription',
     });
   }
 
-  let order = null;
   const query = {
     where: {
       id: orderId,
     },
     include: [
       { model: models.Subscription },
+      { model: models.Tier },
       { model: models.Collective, as: 'collective' },
       { model: models.Collective, as: 'fromCollective' },
     ],
   };
-  return (
-    models.Order.findOne(query)
-      .tap(o => (order = o))
-      .tap(order => {
-        if (!order) {
-          throw new Error('Subscription not found');
-        }
-        return Promise.resolve();
-      })
-      .tap(order => {
-        if (!remoteUser.isAdmin(order.FromCollectiveId)) {
-          throw new errors.Unauthorized({
-            message: "You don't have permission to cancel this subscription",
-          });
-        }
-        return Promise.resolve();
-      })
-      .tap(order => {
-        if (!order.Subscription.isActive && order.status === status.CANCELLED) {
-          throw new Error('Subscription already canceled');
-        }
-        return Promise.resolve();
-      })
-      .then(order => Promise.all([order.update({ status: status.CANCELLED }), order.Subscription.deactivate()]))
-
-      // createActivity - that sends out the email
-      .then(() =>
-        models.Activity.create({
-          type: activities.SUBSCRIPTION_CANCELED,
-          CollectiveId: order.CollectiveId,
-          UserId: order.CreatedByUserId,
-          data: {
-            subscription: order.Subscription,
-            collective: order.collective.minimal,
-            user: remoteUser.minimal,
-            fromCollective: order.fromCollective.minimal,
-          },
-        }),
-      )
-      .then(() => models.Order.findOne(query))
-  ); // need to fetch it second time to get updated data.
+
+  const order = await models.Order.findOne(query);
+
+  if (!order) {
+    throw new Error('Subscription not found');
+  }
+  if (!remoteUser.isAdmin(order.FromCollectiveId)) {
+    throw new errors.Unauthorized({
+      message: "You don't have permission to cancel this subscription",
+    });
+  }
+  if (!order.Subscription.isActive && order.status === status.CANCELLED) {
+    throw new Error('Subscription already canceled');
+  }
+
+  await order.update({ status: status.CANCELLED });
+  await order.Subscription.deactivate();
+  await models.Activity.create({
+    type: activities.SUBSCRIPTION_CANCELED,
+    CollectiveId: order.CollectiveId,
+    UserId: order.CreatedByUserId,
+    data: {
+      subscription: order.Subscription,
+      collective: order.collective.minimal,
+      user: remoteUser.minimal,
+      fromCollective: order.fromCollective.minimal,
+    },
+  });
+
+  if (order.Tier && order.collective.slug === PLANS_COLLECTIVE_SLUG) {
+    const plan = findKey(plans, { slug: order.Tier.slug });
+    if (plan && Boolean(order.fromCollective.plan)) {
+      await order.fromCollective.update({ plan: null });
```

Comment:
We have to think about people upgrading from a plan to another, the best way will be to subscribe to the new plan first, then cancel the old one.

Maybe we should not do that synchronously in the end. Daily cron job and alert us?